### PR TITLE
Allow quoted date in front matter defaults

### PR DESCRIPTION
**Pull request by ducktyper**
Fix to #4068

### CODE REVIEW

**Comment by envygeeks** [comment]
:heart: for doing this @ducktyper

**Comment by ducktyper** [comment]
Yes! that is the better name. Thanks for the quick feedback.

**Comment by parkr** [comment]
This seems like a lack of flexibility on the users' parts. What if a user wants a string there and now we force a date? YAML parses dates just fine: just don't quote it.

**Comment by envygeeks** [comment]
Personally if this were my site I would expect anything labled "date" to be turned into an Object and enforced as so, and if I want a string I would not use an internal front-matter variable that is expected to be a date.

**Comment by ducktyper** [comment]
If I understand the code correctly, in my opinion, I believe the value of "date" from default front-matter and front-matter (of each post or collection file) need to be handled in the same way, because they both can be the value of data["date"] in a Document object which is used as a Time object.

Since Jekyll converts the value of "date" in front-matter to a Time object already in the same way, it is logical to do the same to the default front-matter unless that is not the desirable behaviour.

File: lib/jekyll/frontmatter_defaults.rb [hunk 1]
```diff
@@ -30,6 +30,13 @@ def update_deprecated_types(set)
       set
     end
 
+    def ensure_time!(set)
+      return set unless set.key?('values') && set['values'].key?('date')
+      return set if set['values']['date'].is_a?(Time)
+      set['values']['date'] = Utils.parse_date(set['values']['date'], "Front matter defaults does not have a valid date.")
```
Comment (parkr) [inline lib/jekyll/frontmatter_defaults.rb]: Can you indicate something more helpful in this error message? If you have a 200-line config.yml, this is going to be an annoying error.

**Comment by ducktyper** [comment]
Now the error message shows which front-matter default set to look for by showing the set.
Also found a grammar mistake. Sorry for my unprofessional writing skill.
The invalid date is shown from Utils.parse_date method so I did not need to re-display it.

Here is an example error message that could be generated.

             ERROR: YOUR SITE COULD NOT BE BUILT:
                    ------------------------------------
                    Invalid date '2000-01-0': An invalid date format was found in a front-matter default set: {"scope"=>{"path"=>""}, "values"=>{"layout"=>"post", "date"=>"2000-01-0"}}

Let me know if I missed something else or need more explanation.
Thanks for your feedback.

**Comment by ducktyper** [comment]
I want to fix the bad test but could you enlighten me with little more detail?
I reviewed the tests and came with some ideas but I'm guessing I missed  something else.
1. Using .any? is more readable than .find with "assert" method because  .any? returns true/false while .find returns page/nil
2. Tests "not raise error" and "parse date" can be merged to reduce testing time since @site.process method takes some time to run.

**Pull request merged**